src/media/inner.rs

```diff
@@ -1065,53 +1106,43 @@ impl MediaInner {
     ///
     /// For the purposes of pacing each media is treated as its own packet queue. Internally this
     /// is spread across many [`PacketizingBuffer`] which is where the actual packets are queued.
-    pub fn buffers_tx_queue_state(&self, now: Instant) -> QueueState {
-        let kind = if self.kind() == MediaKind::Audio {
-            MediaKind::Audio
-        } else {
-            MediaKind::Video
-        };
-        let mut state = self
-            .buffers_tx
-            .values()
-            .fold(QueueState::new(kind), |mut state, b| {
-                state.merge(&b.queue_state(now));
-
-                state
-            });
+    pub fn buffers_tx_queue_state(&mut self, now: Instant) -> QueueState {
+        // If it is cached, we don't need to recalculate it.
+        if let Some(queue_state) = self.queue_state {
+            return queue_state;
+        }
 
-        state.id = QueueId::new(self.index() as u64);
+        let init = QueueSnapshot::default();
+        let snapshot = self.buffers_tx.values_mut().fold(init, |mut snap, b| {
+            snap.merge(&b.queue_snapshot(now));
 
-        for resend in &self.resends {
-            let Some(original_packet) = packet_for_resend(&self.buffers_tx, resend) else {
-                continue;
-            };
+            snap
+        });
 
-            state.packet_count += 1;
-            state.size += original_packet.data.len().into();
-            state.total_queue_time += now.saturating_duration_since(resend.queued_at);
-            state.update_leading_queue_time(Some(resend.queued_at));
-        }
+        let state = QueueState {
+            mid: self.mid,
+            is_audio: self.kind == MediaKind::Audio,
+            use_for_padding: self.has_tx_rtx(),
```

Comment:
This is a potential regression, we don't verify that `self.kind == MediaKind::Video`, might be moot point since we don't generally have RTX for audio.

Reply:
Yeah. Let's fix that.